omise-woocommerce.php

```diff
@@ -29,48 +29,90 @@ class Omise {
 	 */
 	protected static $the_instance = null;
 
+	/**
+	 * @since 3.3
+	 *
+	 * @var   boolean
+	 */
+	protected static $can_initiate = false;
+
 	/**
 	 * @since  3.0
 	 */
 	public function __construct() {
-		$this->initiate();
+		add_action( 'plugins_loaded', array( $this, 'check_dependencies' ) );
+		add_action( 'init', array( $this, 'init' ) );
 
 		do_action( 'omise_initiated' );
 	}
 
 	/** 
-	 * Callback to display message about activation error
+	 * Check if all dependencies are loaded
+	 * properly before Omise-WooCommerce.
 	 * 
 	 * @since  3.2
 	 */
-	public function woocommerce_plugin_notice(){
-		?>
-		<div class="error">
-			<p><?php echo __( 'Plugin <strong>deactivated</strong>. The Omise WooCommerce plugin requires <strong>WooCommerce</strong> to be installed and active.', 'omise' ); ?></p>
-		</div>
-		<?php
+	public function check_dependencies() {
+		if ( ! function_exists( 'WC' ) ) {
+			return;
+		}
+
+		static::$can_initiate = true;
 	}
 
-	/** 
-	 * Callback checking if WooCommerce is active
-	 * 
-	 * @since  3.2
+	/**
+	 * @since  3.0
 	 */
-	public function check_woocommerce_active() {
-		if ( is_admin() && current_user_can( 'activate_plugins' ) && !is_plugin_active( 'woocommerce/woocommerce.php' ) ) {
-			add_action( 'admin_notices', array($this, 'woocommerce_plugin_notice') );
-			deactivate_plugins( plugin_basename( __FILE__ ) ); 	
+	public function init() {
+		if ( ! static::$can_initiate ) {
+			add_action( 'admin_notices', array($this, 'init_error_messages') );
+			return;
 		}
+
+		$this->define_constants();
+		$this->include_classes();
+		$this->load_plugin_textdomain();
+		$this->register_post_types();
+		$this->init_admin();
+		$this->init_route();
+
+		register_omise_alipay();
+		register_omise_creditcard();
+		register_omise_internetbanking();
+		prepare_omise_myaccount_panel();
 	}
 
 	/**
-	 * @since  3.0
+	 * Callback to display message about activation error
+	 *
+	 * @since  3.2
+	 */
+	public function init_error_messages(){
+		?>
+		<div class="error">
+			<p><?php echo __( 'Omise WooCommerce plugin requires <strong>WooCommerce</strong> to be activated.', 'omise' ); ?></p>
```

Comment:
Need to add 'The'

Reply:
👍 This will be automatically solved after merge #94